empty_or_echo.sh

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env sh
```

Comment:
This looks awkward. Why not use `echo foo | test ! -s /dev/stdin` at the call sites instead?